catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/history/HistorianTest.java

```diff
@@ -0,0 +1,462 @@
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.operation.impl.UpdateImpl;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.Subject;
+
+public class HistorianTest {
+
+    private static final String METACARD_ID = "METACARD_ID";
+
+    private static final String RESOURCE_URI = "content:example.com";
+
+    private static final String UPDATE_DESCRIPTION = "This is an updated description.";
+
+    private CatalogProvider catalogProvider;
+
+    private InMemoryStorageProvider storageProvider;
+
+    private Historian historian;
+
+    @Before
+    public void setup() {
+        historian = new Historian();
+
+        catalogProvider = mock(CatalogProvider.class);
+        historian.setCatalogProviders(Collections.singletonList(catalogProvider));
+
+        storageProvider = new InMemoryStorageProvider();
+        historian.setStorageProviders(Collections.singletonList(storageProvider));
+
+        historian.setFilterBuilder(new GeotoolsFilterBuilder());
+
+        historian.setMetacardTypes(Collections.singletonList(BasicTypes.BASIC_METACARD));
+
+        Security security = mock(Security.class);
+        Subject subject = mock(MockSubject.class);
+        when(subject.execute(any(Callable.class))).thenCallRealMethod();
+        when(security.getSystemSubject()).thenReturn(subject);
+        historian.setSecurity(security);
+    }
+
+    @Test
+    public void testDoSkip() throws SourceUnavailableException, IngestException {
+        historian.setHistoryEnabled(false);
+        historian.version(mock(UpdateResponse.class));
+
+        verifyZeroInteractions(catalogProvider);
+    }
+
+    @Test
+    public void testSetSkipFlag() throws SourceUnavailableException, IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+        UpdateResponse updateResponse = createUpdateResponse(properties);
+
+        historian.version(updateResponse);
+
+        assertThat(properties, hasEntry(MetacardVersion.SKIP_VERSIONING, true));
+    }
+
+    @Test
+    public void testSkipProperty() throws SourceUnavailableException, IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+        properties.put(MetacardVersion.SKIP_VERSIONING, true);
+
+        UpdateResponse updateResponse = createUpdateResponse(properties);
+
+        historian.version(updateResponse);
+        // Called once for skip check
+        verify(updateResponse).getProperties();
+    }
+
+    @Test
+    public void testUpdateResponse() throws Exception {
+        UpdateResponse updateResponse = createUpdateResponse(null);
+        List<Update> updateList = createUpdatedMetacardList();
```

Comment:
Added deleted and non versioned metacards to the mock query function, this method is used to just get an updated pair of metacards.